### PR TITLE
Add extra metadata information to Atlas pushes

### PR DESCRIPTION
**Pull request by sethvargo**
- [x] Tests

### CODE REVIEW

File: plugins/commands/push/command.rb [hunk 1]
```diff
@@ -9,8 +9,17 @@ def self.synopsis
...
+
         opts = OptionParser.new do |o|
           o.banner = "Usage: vagrant push [strategy] [options]"
+          o.separator ""
+          o.separator "Options:"
+          o.separator ""
+
+          o.on("-m", "--message BODY", "Text to identify this push") do |m|
+            options[:message] = m
+          end
```
Comment (sethvargo) [inline plugins/commands/push/command.rb]: @mitchellh I added this as an option, but I'm not sure how to pass the option into the actual push class. Thoughts?

File: plugins/pushes/atlas/push.rb [hunk 2]
```diff
@@ -27,8 +27,10 @@ def execute(uploader)
         cmd << "-vcs" if config.vcs
         cmd += config.includes.map { |v| ["-include", v] }
         cmd += config.excludes.map { |v| ["-exclude", v] }
+        cmd += metadata.map { |k,v| ["-metadata", "#{k}=#{v}"] }
         cmd += ["-address", config.address] if config.address
         cmd += ["-token", config.token] if config.token
+        cmd += ["-message", message] if message
```
Comment (sethvargo) [inline plugins/pushes/atlas/push.rb]: Re: my previous comment, I need to get the value for that option in here somehow...

File: plugins/commands/push/command.rb (hunk 1 above)
Comment (mitchellh) [inline plugins/commands/push/command.rb]: I also wonder if this should even be a top-level option, because what does it mean in the general sense?

**Comment by mitchellh** [comment]
LGTM

**Pull request merged**